From: Shimazu <vanessimbolon2020@gmail.com>
Date: Fri, 31 Jul 2026 02:19:04 +0000
Subject: ExtinguishEffect: Remove SchedulerUtils, migrate to EntityScheduler

AE PATCH REASON: SchedulerUtils bypasses Folia scheduler lifecycle; obfuscated variable names reduce maintainability.
AE PATCH FIX: Remove SchedulerUtils import, replace with ASManager import. Replace SchedulerUtils.runTaskLater() with target.getScheduler().runDelayed(). Rename var1/var2/var3 to meaningful parameter names.
AE PATCH RISK: LOW

---
diff --git a/net/advancedplugins/ae/impl/effects/effects/effects/internal/ExtinguishEffect.java b/net/advancedplugins/ae/impl/effects/effects/effects/internal/ExtinguishEffect.java
--- a/net/advancedplugins/ae/impl/effects/effects/effects/internal/ExtinguishEffect.java	2026-07-30 10:27:31.667350114 +0000
+++ b/net/advancedplugins/ae/impl/effects/effects/effects/internal/ExtinguishEffect.java	2026-07-30 10:28:05.861658383 +0000
@@ -2,18 +2,18 @@
 
 import net.advancedplugins.ae.impl.effects.effects.actions.execution.ExecutionTask;
 import net.advancedplugins.ae.impl.effects.effects.effects.AdvancedEffect;
-import net.advancedplugins.ae.impl.utils.SchedulerUtils;
+import net.advancedplugins.ae.impl.utils.ASManager;
 import org.bukkit.entity.LivingEntity;
 import org.bukkit.plugin.java.JavaPlugin;
 
 public class ExtinguishEffect extends AdvancedEffect {
-   public ExtinguishEffect(JavaPlugin var1) {
-      super(var1, "EXTINGUISH", "Extinguish an entity", "%e");
+   public ExtinguishEffect(JavaPlugin plugin) {
+      super(plugin, "EXTINGUISH", "Extinguish an entity", "%e");
    }
 
    @Override
-   public boolean executeEffect(ExecutionTask var1, LivingEntity var2, String[] var3) {
-      SchedulerUtils.runTaskLater(() -> var2.setFireTicks(0), 1L);
+   public boolean executeEffect(ExecutionTask task, LivingEntity target, String[] args) {
+      target.getScheduler().runDelayed(ASManager.getInstance(), t -> target.setFireTicks(0), null, 1L);
       return true;
    }
 }
